src/iapetos/collector/ring.clj

```diff
@@ -142,14 +142,21 @@
 ;; ### Metrics Endpoint
 
 (defn wrap-metrics-expose
-  "Expose Prometheus metrics at the given constant URI using the text format."
+  "Expose Prometheus metrics at the given constant URI using the text format.
+
+   If `:on-request` is given, it will be called with the collector registry
+   whenever a request comes in. This lets you use the Prometheus scraper as
+   a trigger for metrics collection."
   [handler registry
-   & [{:keys [path]
+   & [{:keys [path on-request]
        :or {path "/metrics"}}]]
   (fn [{:keys [request-method uri] :as request}]
     (if (= uri path)
       (if (= request-method :get)
-        (metrics-response registry)
+        (do
+          (if on-request
+            (on-request registry))
```

Comment:
I feel like that is not an improvement.
How about binding it to `identity` by default (33d976b)?